### PR TITLE
VZ-9126: Initial commit for OCNE CAPI controllers

### PR DESCRIPTION
**Pull request by theabmitra**
Initial commit of CAPI OCNE bootstrap and control plane. 
- New Proxy resource was introduced 
- New CRDS introduced
```
ocneconfigs                                    bootstrap.cluster.x-k8s.io/v1beta1        true         OCNEConfig
ocneconfigtemplates                            bootstrap.cluster.x-k8s.io/v1beta1        true         OCNEConfigTemplate
ocnecontrolplanes                 ocnecp       controlplane.cluster.x-k8s.io/v1beta1     true         OCNEControlPlane
ocnecontrolplanetemplates                      controlplane.cluster.x-k8s.io/v1beta1     true         OcneControlPlaneTemplate
```

- Sample OCNE Control Plane
 
```
apiVersion: controlplane.cluster.x-k8s.io/v1beta1
kind: OCNEControlPlane
metadata:
  name: ${CLUSTER_NAME}-control-plane
  namespace: ${NAMESPACE}
spec:
  controlPlaneConfig:
    clusterConfiguration:
      apiServer:
        certSANs:
          - localhost
          - 127.0.0.1
      dns:
        imageRepository: foo
        imageTag: 1.8.6
      etcd:
        local:
          imageRepository: foo
          imageTag: 3.5.3
      kubernetesVersion: ${KUBERNETES_VERSION}
      networking: {}
      scheduler: {}
      imageRepository: foo
    initConfiguration:
      nodeRegistration:
        criSocket: /var/run/crio/crio.sock
        kubeletExtraArgs:
          cloud-provider: external
          provider-id: oci://{{ ds["id"] }}
    joinConfiguration:
      discovery: {}
      nodeRegistration:
        criSocket: /var/run/crio/crio.sock
        kubeletExtraArgs:
          cloud-provider: external
          provider-id: oci://{{ ds["id"] }}
    proxy: 
      httpProxy: "http://www-foo.com:80"
      httpsProxy: "http://www-foo.com:80"
      noProxy: "${POD_CIDR},${CLUSTER_CIDR}"
  machineTemplate:
    infrastructureRef:
      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
      kind: OCIMachineTemplate
      name: ${CLUSTER_NAME}-control-plane
      namespace: ${NAMESPACE}
  replicas: ${CONTROL_PLANE_MACHINE_COUNT}
  version: ${KUBERNETES_VERSION}
```

- Sample OCNE Config Template 

```
apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
kind: OCNEConfigTemplate
metadata:
  name: ${CLUSTER_NAME}-md-0
  namespace: ${NAMESPACE}
spec:
  template:
    spec:
      joinConfiguration:
        nodeRegistration:
          kubeletExtraArgs:
            cloud-provider: external
            provider-id: oci://{{ ds["id"] }}
      proxy: 
        httpProxy: "http://www.foo..com:80"
        httpsProxy: "http://www.foo.com:80"
        noProxy: "${POD_CIDR},${CLUSTER_CIDR}"  
```

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+This is derivative work from cluster-api community (https://github.com/kubernetes-sigs/cluster-api)
```
Comment (kminder) [inline NOTICE.txt:4]: s/derivative work/derivative of work

File: NOTICE.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
```
Comment (kminder) [inline NOTICE.txt:1]: Replace this with the Kubernetes copyright and ASLv2 header.  I don't see any point in changing either for this.

File: THIRD_PARTY_LICENSES.txt
Comment (kminder) [inline THIRD_PARTY_LICENSES.txt:1]: Where did this content come from.  Specifically did it come from the BA?

**Comment by kminder** [comment]
I think that every single file copied from https://github.com/kubernetes-sigs/cluster-api might need to have an attribution statement at the top.  Otherwise I'm certain that some files will eventually be changed and this clause from [ASLv2](https://www.apache.org/licenses/LICENSE-2.0) will be forgotten.
> 4b. You must cause any modified files to carry prominent notices stating that You changed the files; and

File: NOTICE.txt (hunk 1 above)
Comment (theabmitra) [reply NOTICE.txt]: done

File: NOTICE.txt (hunk 2 above)
Comment (kminder) [reply NOTICE.txt:1]: I don't think we should be changing the licensing of anything in this repo.

Comment (theabmitra) [reply NOTICE.txt:1]: i am not clear, we are not changing anything with respect to this file. This is a new top level notice file.i added is this not needed ?

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**